Lib/urllib/parse.py

```diff
@@ -60,6 +67,30 @@
                'https', 'shttp', 'rtsp', 'rtspu', 'sip', 'sips',
                'mms', 'sftp', 'tel']
 
+
+def _scheme_classes(scheme, overrides=set()):
+    """Find out what scheme classes a given scheme fits in.
+
+    This consults the variables uses_relative, uses_netloc, and
+    uses_params. It returns a set of all the classes that apply, with
+    at least the unique classes specified by the optional overrides
+    parameter.
+
+    """
+    scheme_classes = set(overrides)
+
+    if scheme in uses_relative:
+        scheme_classes.add(SchemeClass.RELATIVE)
```

Comment:
`options |= SchemeFlag.RELATIVE`

same transformation for the next two branches